stanford/roles/insights_deployment/tasks/main.yml

```diff
@@ -0,0 +1,159 @@
+        cidr_ip:  "0.0.0.0/0"
+      - proto:  "tcp"
+        from_port:  "8443"
+        to_port:  "8443"
+        cidr_ip:  "204.246.160.32/28"
+      - proto:  "tcp"
+        from_port:  "8443"
+        to_port:  "8443"
+        cidr_ip:  "204.246.162.32/28"
+      - proto:  "tcp"
+        from_port:  "8443"
+        to_port:  "8443"
+        cidr_ip:  "54.240.198.0/24"
+      - proto:  "udp"
+        from_port:  "0"
+        to_port:  "65535"
+        group_name:  "{{ env }}-group-emr_master"
+      - proto:  "icmp"
+        from_port:  "0"
+        to_port:  "65535"
+        group_name:  "{{ env }}-group-emr_master"
+  register:  "my_ElasticMapReduce-master_sg"
+
+- name:  Create ElasticMapReduce-slave SG
+  ec2_group:
+    name:  "{{ env }}-group-emr_slave"
+    description:  "Slave group for Elastic MapReduce"
+    vpc_id:  "{{ vpc_id }}"
+    region:  "{{ aws_region }}"
+    rules:
+      - proto:  "tcp"
+        from_port:  "0"
+        to_port:  "65535"
+        group_name:  "{{ env }}-group-emr_slave"
+      - proto:  "tcp"
+        from_port:  "0"
+        to_port:  "65535"
+        group_name:  "{{ env }}-group-emr_master"
+      - proto:  "udp"
+        from_port:  "0"
+        to_port:  "65535"
+        group_name:  "{{ env }}-group-emr_slave"
+      - proto:  "udp"
+        from_port:  "0"
+        to_port:  "65535"
+        group_name:  "{{ env }}-group-emr_master"
+      - proto:  "icmp"
+        from_port:  "0"
+        to_port:  "65535"
+        group_name:  "{{ env }}-group-emr_master"
+      - proto:  "icmp"
+        from_port:  "0"
+        to_port:  "65535"
+        group_name:  "{{ env }}-group-emr_slave"
+  register:  "my_ElasticMapReduce-slave_sg"
+
+- name:  Add slave rules to master SG
+  ec2_group:
+    name:  "ElasticMapReduce-master"
+    purge_rules:  no
+    rules:
+      - proto:  "tcp"
+        from_port:  "0"
+        to_port:  "65535"
+        group_name:  "{{ env }}-group-emr_slave"
+      - proto:  "udp"
+        from_port:  "0"
+        to_port:  "65535"
+        group_name:  "{{ env }}-group-emr_slave"
+      - proto:  "icmp"
+        from_port:  "0"
+        to_port:  "65535"
+        group_name:  "{{ env }}-group-emr_slave"
+
+# Create Mysql Security Group
+- name:  Create MYSQL SG
+  ec2_group:
+    name:  "{{ env }}-analytics-mysql"
+    description:  "{{ env }}-analytics-mysql"
+    vpc_id:  "{{ vpc_id }}"
+    region:  "{{ aws_region }}"
+    rules:
+      - proto:  "tcp"
+        from_port:  "0"
+        to_port:  "65535"
+        group_name: "{{ COMMON_DEPLOYMENT }}-group-{{ CLUSTER_NAME }}"
+      - proto:  "tcp"
+        from_port:  "3306"
+        to_port:  "3306"
+        group_name:  "{{ env }}-group-emr_master"
+      - proto:  "tcp"
+        from_port:  "3306"
+        to_port:  "3306"
+        group_name:  "{{ env }}-group-emr_slave"
+      - proto:  "icmp"
+        from_port:  "0"
+        to_port:  "65535"
+        group_name:  "{{ COMMON_DEPLOYMENT }}-group-{{ CLUSTER_NAME }}"
```

Comment:
Ah, this might explain why you were foreseeing "dependency issues".
This is one of the reasons for the "one cluster/az per subnet" configuration used elsewhere.

By restricting based on network addresses,
the dependents (here, the subnet/vpc and its CIDR block)
don't have to actually exist at the time you create the security group.
You don't get that flexibility when using `group_name`s.

It helps if you think of every "service" as a cluster;
- mysql
- emr_master
- emr_slave

and then whitelist their address spaces accordingly.